audit-log-api/src/use-cases/GetLastEventUseCase.ts

```diff
@@ -1,20 +1,24 @@
 import type { AuditLogDynamoGateway, BichardAuditLogEvent, PromiseResult } from "shared"
 import { isError } from "shared"
 
-export default class GetLastEventUseCase {
+export default class GetLastFailedMessageEventUseCase {
   constructor(private readonly auditLogDynamoGateway: AuditLogDynamoGateway) {}
 
   async get(messageId: string): PromiseResult<BichardAuditLogEvent> {
-    const events = await this.auditLogDynamoGateway.fetchEvents(messageId)
+    const events = (await this.auditLogDynamoGateway.fetchEvents(messageId)) as BichardAuditLogEvent[]
```

Comment:
Would this be `as unknown` until the validation is done?

Reply:
In order to do the validation, we need to access/check some of the properties in the `BichardAuditLogEvent` type - which is why I made this change (previously it was performing the cast at the end of the function, as part of the `return` statement).

I agree this change is a bit mucky; I think really the `auditLogDynamoGateway` needs to be updated to return the correct type - or at least provide an alternative function that validates the type and returns it. However, I think that could turn into a much bigger change in its own right, as that library is shared between all the other audit logging components.